src/VkBootstrap.cpp

```diff
@@ -1970,6 +1978,16 @@ SwapchainBuilder& SwapchainBuilder::set_image_array_layer_count(uint32_t array_l
 	info.array_layer_count = array_layer_count;
 	return *this;
 }
+SwapchainBuilder& SwapchainBuilder::set_desired_min_image_count(uint32_t min_image_count) {
+	info.min_image_count = min_image_count;
+	info.add_to_min_image_count = 0; // Unnecessary, but let's do it for clarity
+	return *this;
+}
```

Comment:
I think this interface would be a lot cleaner if instead of specifying the exact image count, there was an enum that controlled whether to double or triple buffer. 
Is it realistic that there is a need to control anything beyond setting it to one or two?
The presentation engine is pretty free to 'choose' any number higher that minImageCount after all.

Reply:
If this was an enum, it would be a 1-to-1 mapping to an image count, and there is no risk that the mapping may change, so at that point one might as well use the integer directly.
To me, it is as if we defined an enum such as "INT_0", "INT_1" and so on for each integer, just to force the use of a name rather than an integer which, at the end, is the value actually used.

I agree that enums are more type-safe and improve readability in general, however in this particular case I find it less fitting : 
- "min_image_count = 2" reads the same to me as "desire a double buffer";
- Plain integers provide a strict superset of the functionality of the enum; it's very likely that few people want to go beyond triple buffering, but we also have to reason to introduce an artificial limitation that prevents them from doing so.

In any case, the real thing that happens at the end is that we set `create_info.minImageCount`, and I would expect the API to match this closely.
I think `set_desired_min_image_count(N)` tells the reader evertyhing they need to know already: you'd like at least N images, but the presentation engine may provide more. I think an API such as `set_desired_multiple_buffering(DOUBLE_BUFFER)` is at best equivalent, but at worst brings less value.

That's my opinion at least.

Reply:
I really would prefer an enum here for several reasons:

1. Much clearer to everyone. Double and Triple buffering are established terms. `set_swapchain_buffer_mode(DOUBLE_BUFFER)` is much more immediately obvious than `set_desired_min_image_count()`. Part of vk-bootstraps design was to make it "easier" for people new to the API to get up and running. Its a fine line using the correct terminology vs using an abstracted form that signals intent better. In other words, I disagree that `set_desired_min_image_count(N)` has enough information for a user to intuit what it does.

2. I am unaware of any practical use cases for setting it above 3. This is the first issue for vk-bootstrap discussing this count, which I interpret as most people not being that particular (and that triple buffering is a good default).

3. There is an information gap. If users want to reliably set the minImageCount to be triple or double buffered, they first need to know whether the presentation engine supports it. Yet I don't believe there is a mechanism for that currently in vk-bootstrap. `VkSurfaceCapabilitiesKHR.minImageCount` only requires that the value be 1 or greater. So to know what value to use, you'd need to first know what the minImageCount is. This isn't an issue with the direct value setting with `set_desired_min_image_count()` but is a problem with `use_default_min_image_count(add_to_min_image_count)` when `add_to_min_image_count` isn't zero. 

There is a design issue of whether to clamp or return an error if `set_desired_min_image_count()` can't be satisfied. Should vk-bootstrap silently triple buffer when the user asked for double buffering? (Yes, this applies to the enum solution, so it needs to be decided in either case). I lean towards just clamping it as its better to have a working swapchain than one which may not be 'perfect'. If doing exactly what they want is desired, they wouldn't likely be using vk-bootstrap.

Reply:
Why not have a `set_desired_min_image_count` and a `set_required_min_image_count` API? If a required min image count is provided and the surface doesn't satisfy that constraint, it should return an error.  If a required min image count isn't set and a desired is and the surface doesn't satisfy the desired constraint, clamp and return a swapchain.

As for DOUBLE_BUFFER and TRIPLE_BUFFER, how about providing named constants that are just integer values for 2 and 3?